config/dbstructure.config.php

```diff
@@ -1365,7 +1365,7 @@
 		"comment" => "Background tasks queue entries",
 		"fields" => [
 			"id" => ["type" => "int unsigned", "not null" => "1", "extra" => "auto_increment", "primary" => "1", "comment" => "Auto incremented worker task id"],
-			"parameter" => ["type" => "mediumblob", "comment" => "Task command"],
```

Comment:
Why have you changed it to "mediumtext"? The "blob" is done by intention, since it doesn't convert charsets.

Reply:
Because we aren't storing binary data in this field. Like you said, it's JSON, so it's text. We're using utf8mb4 encoding anyway, so I don't think character representation would be a problem.

I left mediumblob for the `photo` and `attach` table because we are storing binary data in those tables.

Reply:
JSON is some kind of binary data as well, since it mustn't be altered concerning the charset in any case. And it is slower and uses more disc space :-)

Reply:
If you store it in utf8mb4 and retrieve it in utf8mb4, there is no charset alteration. Besides, it doesn't seem to contain any remote-submitted data? It's mostly a list of simple strings and integers.

Reply:
> we need some long field there and it has to be a primary index

What prevented you from using a text field?

Reply:
Text fields cannot be primary index fields.

Especially for index fields that are meant for table relations, we really should use ```varbinary``` in the future, since this speeds up the index stuff. (There is no need to internally convert ```ä``` into ```a``` or ```ae``` - depending on your collation)

To make a difference between text fields like the contact name, item title or item body and technically text fields like the url fields or network fields, I would like to change most of these fields in the future. In the last versions I already converted several of these fields (have a look at the ```varbinary``` fields that are already in the database.

Changing this back would be contra productive.

Reply:
My main gripe was that binary fields are hard to handle in phpMyAdmin, they don't display as text even when there's only text in the field and I'm not sure my fix for the worker queue described here would work with a binary field: https://github.com/friendica/friendica/pull/6485#issuecomment-456285848

My intuition is that you're trying to fasten a screw using a hammer. Sure, it will work, but ultimately it isn't the best tool for the job. URLs shouldn't be used as a PRIMARY KEY (use an integer id for that), and it sure as hell isn't a binary string even if it can be coerced into one.

My suggestion:
- Use a regular text field to hold URLs since they are text.
- Use an auto incremented integer for primary key.
- Use a fixed length field (that can be binary!) holding a hash of the URL (or the nURL if it comes down to that).

This is the right thing to do.

Reply:
There are always different ways. This hash thing has a huge disadvantage: You need a good hash to avoid collisions. Also you always need an index on the url as well - since you search for it. But this index couldn't be that long - which would lead to more table scans, which also leads to a lower performance.

Especially on side tables the thing with the auto increment integer does make much problems - I had worked with too much side effects of this at Friendica. Also this hash thingy doesn't work so well. This is my experience from the ```item-content``` table reworks.

I have a suggestion: Can we discuss this point later - at a time when the develop branch is stable again? Until this you could redo this single database field.

In an upcoming discussion we could take more time for this and hopefully others would tell their standpoint as well. Then we had much more time for reasonable decisions. Here we really should go further with the bugfixing and normally shouldn't discuss over such details.

Reply:
Ok to talk about the rest later, I still don't want to undo this field change, like I said I'm not sure my fix will work on a binary field, and besides it is used neither as a primary key nor unique index.